pkg/sriov/sriov.go

```diff
@@ -289,6 +292,12 @@ func (s *sriovManager) FillOriginalVfInfo(conf *sriovtypes.NetConf) error {
 		return fmt.Errorf("failed to find vf %d", conf.VFID)
 	}
 	conf.OrigVfState.FillFromVfInfo(vfState)
+
+	if len(pfLink.Attrs().Vfs) < conf.VFID {
+		return fmt.Errorf("wrong VF ID number of vfs for PF %d, VF ID requested %d", len(pfLink.Attrs().Vfs), conf.VFID)
+	}
+	conf.OrigVfState.EffectiveMAC = pfLink.Attrs().Vfs[conf.VFID].Mac.String()
```

Comment:
im not a fan of setting admin MAC as effective mac here.
we should try to get mac from conf.HostIFName if it exist (string not empty)

then deal with EffectiveMAC being empty at upper layer (if needed, although i dont think we need to worry about it for exposing MAC address in network status)


also `VfState.FillFromVfInfo` does not assume the slice index corresponds to VF ID

Reply:
you are right I change the code